### PR TITLE
Actions precommit

### PR DESCRIPTION
**Pull request by kaitj**
## Proposed changes

This PR performs the following:

* Add a github actions for linting (quality) using `poe` tasks
* Sync pre-commit hooks with github actions
* Add flake8 with the caveat of ignoring E501 (lines longer than 79 characters) to avoid conflict with previous steps
* Perform quality fixes to files

Have run both the pre-commit and the poe quality task to ensure they are lined up.

## Types of changes

What types of changes does your code introduce? Put an `x` in the boxes that apply

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionalitiy)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (if none of the other choices apply)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you are unsure about any of the choices, don't hesitate to ask!

- [x] Changes have been tested to ensure that fix is effective or that a feature works.
- [ ] Changes pass the unit tests
- [x] Code has been run through the `poe quality` task
- [x] I have included necessary documentation or comments (as necessary)
- [ ] Any dependent changes have been merged and published

## Notes
All PRs will undergo the unit testing before being reviewed. You may be requested to explain or make additional changes before the PR is accepted.

### CODE REVIEW

File: labelmerge/workflow/scripts/label_split.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def label_split(atlas_path, output_dir, smk_wildcards, bids_dir):
             label_desc = atlas_metadata[atlas_metadata["Index"] == int(label)][
                 "BIDS Name"
             ].values[0]
-        except:
+        except:  # noqa: E722
```
Comment (tkkuehn) [inline labelmerge/workflow/scripts/label_split.py:79]: I think this check is actually valid -- we should figure out what gets raised if the label is missing, catch that exception specifically, and raise the `ValueError` (we may also want to define a `MetadataError` or something) from that exception.

**Review by tkkuehn — CHANGES_REQUESTED** [review]
One small comment on silencing a pylint error but otherwise looks good

File: labelmerge/workflow/scripts/label_split.py (hunk 1 above)
Comment (kaitj) [reply labelmerge/workflow/scripts/label_split.py]: Fair - I removed this `noqa` and added in the IndexError (due to `values[0]` returning an empty list if `int(label)` does not exist in the dataframe).

I tried to define a MetadataError but was running into a few issues. Can look into defining one in a future commit / separate PR.

**Comment by kaitj** [comment]
fdb9d9c just adds in the action triggers - guess I had made those changes in the web browser.

**Review by tkkuehn — APPROVED** [review]

**Pull request merged**